### PR TITLE
Anonymizing sensitive information within certain fields

### PR DESCRIPTION
**Pull request by ajosh0504**
This PR does the following:
- Adds a "sensitive" field to the config. This can contain a set of fields which cannot be anonymized by traditional faker replacements, but still have sensitive user information that needs to be removed (Eg: process commandline args that can provide useful information about the process but if it contains user directories, we might want to remove them)
- Adds a list of regexes (from the [trufflehog project](https://github.com/dxa4481/truffleHogRegexes/blob/master/truffleHogRegexes/regexes.json)) which are used to drop fields containing secrets, from the list of sensitive fields
-  Adds a "test_configs" folder which consists of configs and things to be set in  `anonymizer.py` (provider_map, high_cardinality_fields, user_regexes), for different types of data users might want to anonymize.

### CODE REVIEW

File: test_configs/windows_process_events/anonymizer_settings.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        "labels.account_id": "self.faker.ssn",
+        "labels.endpoint_id": "self.faker.ssn",
+        "observer.name": "self.faker.word",
+        "user.domain": "self.faker.word",
+        "user.name": "self.faker.word"
+    },
+    "high_cardinality_fields" = {},
+    "user_regexes" = {
+        "regex_1": "(Users)\\\\([^\\\\]+)",
+        "regex_2": "(Users)\\/([^\\/]+)"
```
Comment (christophercutajar) [inline test_configs/windows_process_events/anonymizer_settings.json:16]: Can you please provide a more user-friendly name for these regex

**Review by christophercutajar — COMMENTED** [review]
Apart from the single comment, everything looks good awesome work! Some update to the documentation would be required after this PR.

File: test_configs/windows_process_events/anonymizer_settings.json (hunk 1 above)
Comment (ajosh0504) [reply test_configs/windows_process_events/anonymizer_settings.json]: I have changed them to `user_dir_1` and `user_dir_2` if that helps.

**Review by blaklaybul — APPROVED** [review]
Looks good! I'm happy that we're looking at alternate anonymization methods :)

**Comment by ajosh0504** [comment]
Of course we're still retaining all the work you've done with faker @blaklaybul. This just provides fine grained control on some fields. :)

**Review by christophercutajar — APPROVED** [review]
LGTM 🎉 I think it's better that documents with the keywords are being dropped

**Pull request merged**